### PR TITLE
Add annotated ByteBuffer and Buffer to jdk

### PR DESCRIPTION
**Pull request by NhatDinh**
Please review and let me know if any changes are needed.
@panacekcz @kelloggm

### CODE REVIEW

File: checker/jdk/index/src/java/nio/ByteBuffer.java [hunk 1]
```diff
@@ -0,0 +1,2076 @@
...
+     *
+     * @param  capacity
+     *         The new buffer's capacity, in bytes
+     *
+     * @return  The new byte buffer
+     *
+     * @throws  IllegalArgumentException
+     *          If the <tt>capacity</tt> is a negative integer
+     */
+    public static ByteBuffer allocateDirect(int capacity) {
```
Comment (panacekcz) [inline checker/jdk/index/src/java/nio/ByteBuffer.java]: `capacity` should be non-negative.

File: checker/jdk/index/src/java/nio/Buffer.java [hunk 2]
```diff
@@ -0,0 +1,571 @@
...
+            this.mark = mark;
+        }
+    }
+
+    /**
+     * Returns this buffer's capacity.
+     *
+     * @return  The capacity of this buffer
+     */
+    public final int capacity() {
```
Comment (panacekcz) [inline checker/jdk/index/src/java/nio/Buffer.java]: Annotations are missing in this file.

**Comment by NhatDinh** [comment]
Please review newly added annotations in `Buffer.java`.
@panacekcz @kelloggm

File: checker/jdk/index/src/java/nio/Buffer.java [hunk 3]
```diff
@@ -0,0 +1,574 @@
...
+        return capacity;
+    }
+
+    /**
+     * Returns this buffer's position.
+     *
+     * @return  The position of this buffer
+     */
+    public final @NonNegative @LessThan("this.limit + 1") int position() {
```
Comment (panacekcz) [inline checker/jdk/index/src/java/nio/Buffer.java]: The `@LessThan` annotation references a private field, which will be hard to work with for the users and the checker. It's leaking private implementation details to the public interface. I suggest to remove the `@LessThan` annotations.

File: checker/jdk/index/src/java/nio/Buffer.java [hunk 4]
```diff
@@ -0,0 +1,574 @@
...
+        return this;
+    }
+
+    /**
+     * Returns the number of elements between the current position and the
+     * limit.
+     *
+     * @return  The number of elements remaining in this buffer
+     */
+    public final int remaining() {
```
Comment (panacekcz) [inline checker/jdk/index/src/java/nio/Buffer.java]: Can the result be `@NonNegative`?

File: checker/jdk/index/src/java/nio/Buffer.java [hunk 5]
```diff
@@ -0,0 +1,574 @@
...
+     *
+     * @throws  ReadOnlyBufferException
+     *          If this buffer is backed by an array but is read-only
+     *
+     * @throws  UnsupportedOperationException
+     *          If this buffer is not backed by an accessible array
+     *
+     * @since 1.6
+     */
+    public abstract int arrayOffset();
```
Comment (panacekcz) [inline checker/jdk/index/src/java/nio/Buffer.java]: Can the result be `@NonNegative`?

File: checker/jdk/index/src/java/nio/Buffer.java (hunk 2 above)
Comment (panacekcz) [reply checker/jdk/index/src/java/nio/Buffer.java]: Can the result be `@NonNegative`?

File: checker/jdk/index/src/java/nio/ByteBuffer.java [hunk 6]
```diff
@@ -0,0 +1,2076 @@
...
+     *
+     * @param  capacity
+     *         The new buffer's capacity, in bytes
+     *
+     * @return  The new byte buffer
+     *
+     * @throws  IllegalArgumentException
+     *          If the <tt>capacity</tt> is a negative integer
+     */
+    public static ByteBuffer allocate(int capacity) {
```
Comment (panacekcz) [inline checker/jdk/index/src/java/nio/ByteBuffer.java]: Capacity should be `@NonNegative`.

File: checker/jdk/index/src/java/nio/ByteBuffer.java [hunk 7]
```diff
@@ -0,0 +1,2076 @@
...
+     * @return  The offset within this buffer's array
+     *          of the first element of the buffer
+     *
+     * @throws  ReadOnlyBufferException
+     *          If this buffer is backed by an array but is read-only
+     *
+     * @throws  UnsupportedOperationException
+     *          If this buffer is not backed by an accessible array
+     */
+    public final int arrayOffset() {
```
Comment (panacekcz) [inline checker/jdk/index/src/java/nio/ByteBuffer.java]: Can the result be `@NonNegative`?

**Comment by NhatDinh** [comment]
I've resolved the merge conflicts discussed in our meeting in annotated-libraries. I think this PR is ready for merge.
@panacekcz

**Pull request merged**